$OpenBSD: patch-src_openvpn_ssl_openssl_c,v 1.3 2017/02/06 10:22:35 jca Exp $

Avoid accessing a field of SSL_CTX that is now private in LibreSSL.

--- src/openvpn/ssl_openssl.c.orig	Sat Feb  4 18:30:20 2017
+++ src/openvpn/ssl_openssl.c	Sat Feb  4 18:35:47 2017
@@ -485,14 +485,6 @@ tls_ctx_load_ecdh_params(struct tls_root_ctx *ctx, con
 
     /* Generate a new ECDH key for each SSL session (for non-ephemeral ECDH) */
     SSL_CTX_set_options(ctx->ctx, SSL_OP_SINGLE_ECDH_USE);
-#if OPENSSL_VERSION_NUMBER >= 0x10002000L
-    /* OpenSSL 1.0.2 and newer can automatically handle ECDH parameter loading */
-    if (NULL == curve_name)
-    {
-        SSL_CTX_set_ecdh_auto(ctx->ctx, 1);
-        return;
-    }
-#endif
     /* For older OpenSSL, we'll have to do the parameter loading on our own */
     if (curve_name != NULL)
     {
@@ -502,6 +494,10 @@ tls_ctx_load_ecdh_params(struct tls_root_ctx *ctx, con
     }
     else
     {
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+        SSL_CTX_set_ecdh_auto(ctx->ctx, 1);
+        return;
+#else
         /* Extract curve from key */
         EC_KEY *eckey = NULL;
         const EC_GROUP *ecgrp = NULL;
@@ -519,6 +515,7 @@ tls_ctx_load_ecdh_params(struct tls_root_ctx *ctx, con
         {
             nid = EC_GROUP_get_curve_name(ecgrp);
         }
+#endif
     }
 
     /* Translate NID back to name , just for kicks */
